### PR TITLE
Bring MeanResponseTransformer in line with new test set up

### PR DESCRIPTION
**Pull request by davidhopkinson26**
Bringing MeanResponseTransformer in line with the new testing set up.

Have noticed during this update that WeightColumnFitMixinTests and WeightColumnInitMixinTests might not be being used in other transformers we have already refactored so might be worth a quick audit.

### CODE REVIEW

**Comment by davidhopkinson26** [comment]
In the process of making this PR i created issue #269 as I found the `minimal_dataframe_lookup` method a bit clunky to interact with.

File: tests/nominal/test_MeanResponseTransformer.py [hunk 1]
```diff
@@ -1141,77 +1049,6 @@ def expected_df_10():
...
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.nominal.BaseNominalTransformer,
-            "transform",
-            expected_call_args,
-            return_value=create_MeanResponseTransformer_test_df(),
-        ):
-            x.transform(df)
-
-    def test_learnt_values_not_modified(self):
```
Comment (limlam96) [inline tests/nominal/test_MeanResponseTransformer.py:1195]: Q: how come we're dropping this one?

Comment (davidhopkinson26) [reply tests/nominal/test_MeanResponseTransformer.py:1195]: that was a mistake.  added back in https://github.com/lvgig/tubular/pull/262/commits/2bcaab00452ea73a41d15efc5e75c299eea62ad1

**Comment by limlam96** [comment]
fyi CI is failing, but happy to approve otherwise!

**Comment by davidhopkinson26** [comment]
> fyi CI is failing, but happy to approve otherwise!

Bit strange, an error seems to have appeared with NearestMeanResponseImputer.  Will investigate.

**Comment by davidhopkinson26** [comment]
Had to reconfigure a few things in Imputer tests to make some shared generic tests compatible with this transformer.

- convention now is that the target is column 'a' in the default dataframes.  
- generic tests specify a y argument for fit rather than calling transformer.fit(df)

I'm wondering if it might be best to separate out GenericFitTests for the limited subset of cases where y is a compulsory argument (this and NearestMeanResponse imputer). Quite a few of the generic tests before this PR did not specify a y argument.  I think it is better that they do even though most can take y=None (and in fact default to it) as they will mostly be called in a pipeline where y will be specified so we need to assert they work in those circumstances.  Arguably a test checking those which can accept None work in those circumstances should be created.

**Comment by Chip2916** [comment]
Hi David - picking up this review as Liam is on AL. On your comment regarding transformers where y is not a compulsory argument and a test to confirm they work when None is passed should be created, is there a ticket for this? If not can we get one down :)

**Comment by Chip2916** [comment]
Reviewed scripts, no further comments on Liam's additional review. Agree with the changes made to standardise things across the imputer test scripts

**Comment by davidhopkinson26** [comment]
> Reviewed scripts, no further comments on Liam's additional review. Agree with the changes made to standardise things across the imputer test scripts

https://github.com/lvgig/tubular/issues/287

**Review by Chip2916 — APPROVED** [review]
Happy to approve based on mine and Liam's reviews.

**Pull request merged**